docs/sphinx/_ext/dagster-sphinx/dagster_sphinx/__init__.py

```diff
@@ -73,9 +73,9 @@ def record_error(env: BuildEnvironment, message: str) -> None:
 
 
 def check_public_method_has_docstring(env: BuildEnvironment, name: str, obj: object) -> None:
-    if name != "__init__" and not obj.__doc__:
+    if name != "__init__" and not hasattr(obj, "__doc__"):
         message = (
-            f"Docstring not found for {object.__name__}.{name}. "
+            f"Docstring not found for {obj.__name__}.{name}. "
```

Comment:
```suggestion
            f"Docstring not found for {obj.__name__}.{name}. "  # type: ignore[reportAttributeAccessIssue]
```
Appease the Pyright gods.

Reply:
should actually be repr(obj) not every obj has a name